knative-operator/pkg/common/telemetry/telemetry_metrics.go

```diff
@@ -0,0 +1,25 @@
+package telemetry
+
+import (
+	"github.com/prometheus/client_golang/prometheus"
+	"sigs.k8s.io/controller-runtime/pkg/metrics"
+)
+
+var (
+	serverlessTelemetryG = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Name: "serverless_telemetry",
+			Help: "Reports number of serverless resources for telemetry",
+		},
+		[]string{"type"},
+	)
+	ServicesG  = serverlessTelemetryG.WithLabelValues("service")
+	RoutesG    = serverlessTelemetryG.WithLabelValues("route")
+	RevisionsG = serverlessTelemetryG.WithLabelValues("revision")
+	SourcesG   = serverlessTelemetryG.WithLabelValues("source")
```

Comment:
How do these differ from the metrics that we surface via `GenerateAndServeCRMetrics`?

Reply:
I tried that framework before coming up with this simplified solution. Here are the downsides:
- It requires a cfg object passed by a controller which makes it hard to test.
- it adds a redundant label for the CR type that distinguishes the type of metric [here](https://github.com/operator-framework/operator-sdk/blob/125d0dfcc71fef4f9d7e2a42b1354cb79ffdee03/pkg/kube-metrics/metrics.go#L100), increasing the carnality (max is 10 for telemetry).